### PR TITLE
types: Fix potential timezone related bugs caused by `gotime.Local`

### PR DESCRIPTION
**Pull request by Rustin-Liu**
### What problem does this PR solve? 
#13626 
### What is changed and how it works?

Use sc.TimeZone to create time variable when we round time in RoundFrac.

### Check List 

Tests 

None

Code changes

None

Side effects

None

Related changes

None

Release note

None

### CODE REVIEW

**Comment by Deardrops** [comment]
Thanks for your contribution, before your PR merge, please add some test.

**Comment by Deardrops** [comment]
LGTM, @bb7133 PTAL

File: types/time_test.go [hunk 1]
```diff
@@ -733,6 +733,27 @@ func (s *testTimeSuite) TestRoundFrac(c *C) {
...
+	}
+	// test different time zone
+	losAngelesTz, _ := time.LoadLocation("America/Los_Angeles")
+	sc.TimeZone = losAngelesTz
+	tbl = []struct {
+		Input  string
+		Fsp    int8
+		Except string
+	}{
+		{"2019-11-25 07:25:45.123456", 4, "2019-11-25 07:25:45.1235"},
```
Comment (bb7133) [inline types/time_test.go]: Looks that these cases pass with original `gotime.Local` codes, would you adding some cases that match the changes?

Comment (Rustin-Liu) [reply types/time_test.go]: Added.

**Comment by bb7133** [comment]
Thanks for your help! @Rustin-Liu

**Pull request merged**